.ci/Jenkinsfile

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env groovy
+
+// Verified with Jenkins v2.164.3
+
+// TODO:
+// 1. Calculate taskset/affinity for the scripts based on total number of jenkins executors
+
+pipeline {
+    agent {label "ml-test-node-gpu"}
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'10'))
+        timeout(time: 15, unit: 'MINUTES')
+        disableConcurrentBuilds()
+    }
+
+    environment {
+        ARTEFACT_DIR = "${WORKSPACE}/artefacts"
+        NFS_WORKSPACE = "${NFS_WORKSPACE_ROOT}/ml-nccl-rdma-sharp-plugins-pr/${BUILD_NUMBER}"
+        NCCL_SRC_DIR = "${WORKSPACE}/nccl"
+        NCCL_DIR = "${NFS_WORKSPACE}/nccl"
+        NCCL_TESTS_DIR = "${NFS_WORKSPACE}/nccl-tests"
+    }
+
+    stages {
+        stage('Preparations') {
+            steps {
+                echo 'Preparations...'
+                sh 'mkdir -p ${ARTEFACT_DIR}'
+                sh 'mkdir -p ${NFS_WORKSPACE}'
+            }
+        }
+        stage('Checkout NCCL') {
+            steps {
+                sh 'mkdir -p ${NCCL_SRC_DIR}'
+                dir("${NCCL_SRC_DIR}") {
+                    git branch: 'master',
+                    url: 'git@gitlab.com:Mellanox/nccl-sharp.git'
+                }
+            }
+        }
+        stage('Build NCCL') {
+            steps {
+                echo 'Building NCCL...'
+                sh 'set -o pipefail; ${WORKSPACE}/.ci/build_nccl.sh 2>&1 | tee ${ARTEFACT_DIR}/build_nccl.log'
+                sh 'mkdir -p ${NCCL_DIR}'
+                sh 'cp -R ${NCCL_SRC_DIR}/build/* ${NCCL_DIR}'
+                sh 'find ${NCCL_DIR} -type f'
```

Comment:
why find needed?

Reply:
Just to print ${NCCL_DIR} content into log file (for debug purposes).

Reply:
why not to do it from build_nccl.sh script?
IMO - pipeline should be glue only for scripts, all debug or any other info - print in the script.

Reply:
Moved inside build_nccl.sh (will be printed only if $DEBUG is defined).